source/core/NumericString.cpp

```diff
@@ -817,9 +817,11 @@ void Format(SubString *format, Int32 count, StaticTypeAndData arguments[], Strin
                     case 't':
                     case 'T':
                     {
-                        Int32 tz = Date::getLocaletimeZone();
-                        if (fOptions.EngineerNotation || fOptions.FormatChar == 't') {
-                             tz = 0;
+                        Int32 tz = 0;
+                        if (fOptions.FormatChar == 'T' && !fOptions.EngineerNotation) {
```

Comment:
I am not sure if we want to add an extra check "argType->IsTimestamp()" (or assert/fail inside that about argType)?